File: 10주차/[LV2] 괄호 회전하기.js
// 1차 시도 (성공)
function solution(s) {
	// 올바른 괄호 문자열인지 확인하는 함수
	function isValid(str) {
		const stack = [];
		const opens = "({[";
		const closes = ")}]";

		for (let char of str) {
			if (opens.includes(char)) {
				stack.push(char);
			} else {
				if (stack.length === 0) return false;
				const lastOpen = stack.pop();
				if (opens.indexOf(lastOpen) !== closes.indexOf(char)) return false;
			}
		}

		return stack.length === 0;
	}

	let count = 0;
	const len = s.length;

	// 문자열을 회전시키면서 각 경우를 확인
	for (let i = 0; i < len; i++) {
		const rotated = s.slice(i) + s.slice(0, i);
		if (isValid(rotated)) count++;
	}

	return count;
}

File: 10주차/전진영/[LV2] 괄호 회전하기.py
def solution(s):
    answer=0
    # 현재 문자열이 올바른 괄호인지?
    def isRight(s):
        stack=[]
        
        for char in s:
            if(char=='('or char=='[' or char== '{'):
                stack.append(char)
            else:
                if len(stack)==0: return False
                current=stack.pop()
                if char==')' and current != '(':
                    return False
                elif char=='}' and current!= '{':
                    return False
                elif char==']' and current!= '[':
                    return False
        return len(stack) == 0
        
    
    for x in range(0,len(s)): #0<=x<len(s) 만큼 반복
        #회전을 시키고
        s=s[1:len(s)]+s[0]
        #올바른 괄호열인지 체크
        if isRight(s):
            answer+=1 #올바른 괄호가 되는 횟수를 누적한다.
            
        # print(s,isRight(s))
        
    return answer